python/lsst/ts/ess/common/abstract_command_handler.py

```diff
@@ -78,6 +80,8 @@ def __init__(self, callback: typing.Callable, simulation_mode: int) -> None:
         self._configuration: typing.Optional[typing.Dict[str, typing.Any]] = None
         self._started = False
 
+        self._devices: typing.List[BaseDevice] = []
```

Comment:
Regarding the fact that this constructor is abstract (lines 68 and 69) above):

I suggest making all your `__init__` methods concrete, unless you have the special case that the abstract base class has no other abstract methods, yet you still feel it must be abstract. Making `__init__` abstract is an annoyance because subclasses for which the base class constructor suffices must still define their own.

This particular class has no other abstract methods so it potentially is one of these special cases. But you also have a mock subclass of this class that has no additional behavior, suggesting that this base class could usefully be a _concrete_ class.

You have several other abstract classes that have other abstract methods, and so don't need `__init__` to be abstract.

Reply:
Done. Again, I did this because I am used to the way Java handles abstract classes and prevents initialization even if there are no abstract methods in them. I understand that Python handles this differently so I am adapting my coding style to that.